### PR TITLE
Replace assert with expect

### PR DESCRIPTION
**Pull request by wlin57**
Seems like there will be a small merge conflict (fixed)

### CODE REVIEW

File: test/expect.php [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?hh
+
+namespace Facebook\HackCodegen;
+
+function expect<T>(T $obj, mixed ...$args): CodegenExpectObj<T> {
+  return new CodegenExpectObj(new ImmVector(\func_get_args()));
```
Comment (kmeht) [inline test/expect.php]: Are we not allowed to use Hack arrays here?

File: test/expect.php [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+}
+
+function expect_with_context<T>(string $context, T $obj, mixed ...$args): CodegenExpectObj<T> {
+  return new CodegenExpectObj(new ImmVector(\array_slice(\func_get_args(), 1)), $context);
```
Comment (kmeht) [inline test/expect.php]: ditto

Comment (fredemmott) [reply test/expect.php]: The base class (in another repo) still uses collections (https://github.com/hhvm/fbexpect/blob/master/src/ExpectObj.php)

**Comment by fredemmott** [comment]
@wlin57 : looks like the commit to move the epxect file just removed it, didn't add it

**Comment by fredemmott** [comment]
hmm, the overall branch looks fine, just that commit is weird

**Comment by wlin57** [comment]
hm, the commit itself looks fine on my end.

**Pull request merged**